src/main/java/me/profelements/dynatech/items/electric/AntigravityBubble.java

```diff
@@ -57,48 +63,70 @@ public boolean isSynchronized() {
 
     @Override
     public void tick(Block b) {
+        Set<UUID> tempUuids = new HashSet<>();
+        tempUuids.clear();
         Collection<Entity> bubbledEntities = b.getWorld().getNearbyEntities(b.getLocation(), 25, 25, 25);
-
         for (Entity entity : bubbledEntities) {
             if (entity instanceof Player) {
                 Player p = (Player) entity;
-
+                if (allEnabledPlayers.get(b.getLocation()) == null) {
+                    tempUuids.add(p.getUniqueId());
+                    allEnabledPlayers.put(b.getLocation(), tempUuids);
+                } else {
+                    
+                    allEnabledPlayers.get(b.getLocation()).add(p.getUniqueId());
+                }
                 if (!p.getAllowFlight()) {
-                    enabledPlayers.add(p.getUniqueId());
                     p.setAllowFlight(true);
                     removeCharge(b.getLocation(), getEnergyConsumption());
                 }
             }
         }
 
-        final Iterator<UUID> playerIterator = enabledPlayers.iterator();
-        while (playerIterator.hasNext()) {
-            final UUID uuid = playerIterator.next();
-            Player p = Bukkit.getPlayer(uuid);
 
-            if (p != null && !bubbledEntities.contains(p)) {
-                p.setAllowFlight(false);
-                p.setFlying(false);
-                p.setFallDistance(0.0f);
-                playerIterator.remove();
+
+        if (allEnabledPlayers.get(b.getLocation()) != null) {
+            for (UUID uuid : allEnabledPlayers.get(b.getLocation())) {
+                Player p = Bukkit.getPlayer(uuid);
+                if (p != null && !bubbledEntities.contains(p)) {
+                    allEnabledPlayers.get(b.getLocation()).remove(p.getUniqueId());
+                    checkPlayer(p.getUniqueId());
+                }
             }
         }
     }
+
+    private void checkPlayer(UUID u) {
```

Comment:
Ill look at this method when I'm back at my PC tomorrow and also give this a good general test